clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -440,16 +440,36 @@ private boolean renameObject(LakeFSFileStatus srcStatus, Path dst) throws IOExce
 
         ObjectsApi objects = lfsClient.getObjects();
         //TODO (Tals): Can we add metadata? we currently don't have an API to get the metadata of an object.
-        ObjectStageCreation creationReq = new ObjectStageCreation()
-                .checksum(srcStatus.getChecksum())
-                .sizeBytes(srcStatus.getLen())
-                .physicalAddress(srcStatus.getPhysicalAddress());
+        ObjectCopyCreation creationReq = new ObjectCopyCreation()
+                .srcRef(srcObjectLoc.getRef())
+                .srcPath(srcObjectLoc.getPath());
 
+        Boolean fallbackToStage = false;
         try {
-            objects.stageObject(dstObjectLoc.getRepository(), dstObjectLoc.getRef(), dstObjectLoc.getPath(),
+            objects.copyObject(dstObjectLoc.getRepository(), dstObjectLoc.getRef(), dstObjectLoc.getPath(),
                     creationReq);
         } catch (ApiException e) {
-            throw translateException("renameObject: src:" + srcStatus.getPath() + ", dst: " + dst + ", failed to stage object", e);
+            if (e.getCode() !=HttpStatus.SC_INTERNAL_SERVER_ERROR ||
```

Comment:
**NOT** nit: Is this the only way to detect that copy is not supported?  Because there are many possible causes for ISE, and we will double our API calls for _each one_.

Reply:
The full response on failure is:
```
❯ http get https://treeverse.us-east-1.lakefscloud.io/api/v1/a/b/c
HTTP/1.1 500 Internal Server Error
Connection: keep-alive
Content-Length: 35
Content-Type: application/json
Date: Sun, 15 Jan 2023 07:59:26 GMT
X-Content-Type-Options: nosniff

{
    "message": "invalid API endpoint"
}
```

Is "invalid API endpoint" on the JSON message a part of the API?  (Even if it isn't, it won't change on _old_ server versions, so it's probably safe to use!)